lib/concurrent/channel.rb

```diff
@@ -1,6 +1,272 @@
-require 'concurrent/channel/blocking_ring_buffer'
-require 'concurrent/channel/buffered_channel'
-require 'concurrent/channel/channel'
-require 'concurrent/channel/ring_buffer'
-require 'concurrent/channel/unbuffered_channel'
-require 'concurrent/channel/waitable_list'
+require 'concurrent/channel/buffer'
+require 'concurrent/channel/selector'
+
+require 'concurrent/maybe'
+require 'concurrent/executor/cached_thread_pool'
+
+module Concurrent
+
+  # {include:file:doc/channel.md}
+  class Channel
+    include Enumerable
+
+    GOROUTINES = Concurrent::CachedThreadPool.new
+    private_constant :GOROUTINES
```

Comment:
Absolutely it could. But since this is still early in development I wanted to keep these separate from the global pools, just in case, should channels exhibit any erratic behavior. I discuss this a little in [channel.md](https://github.com/ruby-concurrency/concurrent-ruby/blob/master/doc/channel.md). Should this ever move from Edge to Core I expect that they won't need a separate pool any more.